services/powchain/src/main/java/tech/pegasys/teku/services/powchain/DepositTreeSnapshotConfiguration.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.services.powchain;
+
+import java.util.Optional;
+
+public class DepositTreeSnapshotConfiguration {
+  private final Optional<String> customDepositSnapshotPath;
+  private final Optional<String> checkpointSyncDepositSnapshotUrl;
+  private final Optional<String> bundledDepositSnapshotPath;
+  private final boolean depositSnapshotEnabled;
+
+  DepositTreeSnapshotConfiguration(
+      final Optional<String> customDepositSnapshotPath,
+      final Optional<String> checkpointSyncDepositSnapshotUrl,
+      final Optional<String> bundledDepositSnapshotPath,
+      final boolean depositSnapshotEnabled) {
```

Comment:
you initialize it with `isBundledSnapshotEnabled`, so it's better toc ontinue with this name and rename getter too

Reply:
yeah, this thing is very confusing, if we rename it and getters it would be more clear I think